evm/exceptions.py

```diff
@@ -134,3 +134,11 @@ class OutOfBoundsRead(VMError):
     boundaries of the buffer (such as with RETURNDATACOPY)
     """
     pass
+
+
+class UnannouncedStateAccess(VMError):
```

Comment:
Question about this exception.  *Should* raising this error result in a transaction failing with an OOG error or should it bubble beyond VM execution and cause a hard failure?  I'm assuming the former, but if it is the latter, this should not be a subclass of `VMError` as anything that inherits from that exception class will be captured as part of message execution.

Reply:
You're right, the former. Scenario: An attacker creates a transaction which does some heavy computation and only in the end accesses some state outside of the access list. In that case he needs to pay for the gas, otherwise the collator would have worked for free.